python/shotgun_desktop/paths.py

```diff
@@ -98,11 +99,28 @@ def get_pipeline_configuration_info(connection):
     )
     logger.debug(pprint.pformat(pcs))
 
-    if len(pcs) == 0:
-        pc = None
-    else:
-        # Pick the last result. See the big comment before the Shotgun query to understand.
+    pc = None
+    if len(pcs):
+        # Make subsets of pcs with those that restrict the logged user and those with no restrictions.
+        logged_user_id = sgtk.get_authenticated_user().resolve_entity().get("id")
+        restricted_pcs = []
+        unrestricted_pcs = []
+        for p in pcs:
+            if not p.get("users"):
+                unrestricted_pcs.append(p)
+                continue
+            for restricted_user in p.get("users"):
+                if logged_user_id == restricted_user.get("id"):
+                    restricted_pcs.append(p)
```

Comment:
You could use a generator to avoid this loop here that make the algorithm look complex:

```python
if not p.get("users"):
    unrestricted_pcs.append(p)
elif logged_user_id in [ u.get("id") for u in p.get("users") ]:
    restricted_pcs.append(p)

```